aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,273 @@
+            help="Name of the log output file",
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=str,
+            default=cls._DEFAULT_OUTPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.input_filename",
+            valid_type=str,
+            default=cls._DEFAULT_INPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.scheduler_stdout",
+            valid_type=str,
+            default="_scheduler-stdout.txt",
+            help="Filename to which the content of stdout of the scheduler is written.",
+        )
+        spec.inputs["metadata"]["options"]["parser_name"].default = "janus.parser"
+        spec.inputs.validator = cls.validate_inputs
+        # Define outputs. The default is a dictionary with the content of the xyz file
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        print("defining outputs")
+        spec.output("std_output", valid_type=SinglefileData)
+        spec.output("log_output", valid_type=SinglefileData)
+        spec.output("xyz_output", valid_type=SinglefileData)
+        print("defining outputnode")
+        spec.default_output_node = "results_dict"
+
+        # Exit codes
+
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+        spec.exit_code(
+            306,
+            "ERROR_EMPTY_OUTPUT_FILES",
+            message="File.xyz is empty",
+        )
+        spec.exit_code(
+            307,
+            "ERROR_CONTENT_OUTPUT_FILES",
+            message="The output file does not contain the right content",
+        )
+
+    @classmethod
+    def validate_inputs(
+        cls, inputs: dict, port_namespace: aiida.engine.processes.ports.PortNamespace
+    ) -> Union[str, None]:
+        """
+        Check if the inputs are valid.
+
+        Parameters
+        ----------
+        inputs : dict
+            The inputs dictionary.
+
+        port_namespace : `aiida.engine.processes.ports.PortNamespace`
+            An instance of aiida's `PortNameSpace`.
+
+        Returns
+        -------
+        str or None
+            Error message if validation fails, None otherwise.
+        """
+        # Wrapping processes may choose to exclude certain input ports
+        # If the ports have been excluded, skip the validation.
+        if any(key not in port_namespace for key in ("calctype", "structure")):
+            return None
+
+        for key in ("calctype", "structure"):
+            if key not in inputs:
+                return f"required value was not provided for the `{key}` namespace."
+
+        valid_calctypes = {"singlepoint", "geom opt"}
+        if "calctype" in inputs:
+            if str(inputs["calctype"].value) not in valid_calctypes:
```

Comment:
Don't need to nest `if`s like this. Can combine with `and` instead which is short circuiting, so they are only executed while the previous one is true.
```suggestion
        if (
                "calctype" in inputs and
                str(inputs["calctype"].value) not in valid_calctypes
        ):
```